components/hab/src/cli/hab/util/tls.rs

```diff
@@ -0,0 +1,140 @@
+//! Types for reading certificates, private keys, and root certificate stores from the CLI
+
+use habitat_core::tls;
+use rustls::{Certificate,
+             PrivateKey as RustlsPrivateKey,
+             RootCertStore};
+use serde::{Deserialize,
+            Serialize};
+use std::{path::PathBuf,
+          str::FromStr};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(try_from = "&str", into = "String")]
+pub struct CertificateChain {
+    path:         PathBuf,
+    certificates: Vec<Certificate>,
+}
+
+impl CertificateChain {
+    pub fn certificates(self) -> Vec<Certificate> { self.certificates }
+}
+
+impl FromStr for CertificateChain {
+    type Err = habitat_core::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let path = PathBuf::from(s);
+        let certificates = if path.is_dir() {
+            // TODO: If it is a directory generate a self signed cert and use that or use an
+            // existing cert with a given naming scheme
+            todo!()
+        } else {
+            tls::rustls_reader::certificates_from_file(&path)?
+        };
+        Ok(Self { path, certificates })
+    }
+}
```

Comment:
Yeah, my naming of types was confusing. In https://github.com/habitat-sh/habitat/pull/7923/commits/4dae869ddeacd046adc968a7d5794e4dba528050 I added `Cli` to these type names. Does that help? They have to implement `FromStr` because that is what `structopt` looks for when parsing CLI arguments.

Reply:
Adding `Cli` to the types helps... I think if we add some documentation to drive home the fact that these types are really only intended for codifying the transition from invocation of the CLI to internal types, as well as notes indicating why it's `FromStr` and not `TryFrom<Path>`, that would be good.

Reply:
Agreed. Addressed in the comment [here](https://github.com/habitat-sh/habitat/blob/f35193a5a4b2fcf307ae077696d6ddf17af78459/components/core/src/tls/rustls_wrapper/types.rs#L1)